paddle/fluid/framework/operator.cc

```diff
@@ -835,10 +835,19 @@ static void CheckTensorNANOrInf(const std::string& op_type,
       tensor.type() != proto::VarType::FP64) {
     return;
   }
-  PADDLE_ENFORCE(!framework::TensorContainsInf(tensor),
-                 "Operator %s output Tensor %s contains Inf", op_type, name);
-  PADDLE_ENFORCE(!framework::TensorContainsNAN(tensor),
-                 "Operator %s output Tensor %s contains NAN", op_type, name);
+  if (FLAGS_fast_check_nan_inf) {
+    PADDLE_ENFORCE_NOTNANINF(!framework::TensorContainsInf(tensor),
+                             "Operator %s output Tensor %s contains Inf",
+                             op_type, name);
+    PADDLE_ENFORCE_NOTNANINF(!framework::TensorContainsNAN(tensor),
+                             "Operator %s output Tensor %s contains NAN",
+                             op_type, name);
+  } else {
+    PADDLE_ENFORCE(!framework::TensorContainsInf(tensor),
+                   "Operator %s output Tensor %s contains Inf", op_type, name);
+    PADDLE_ENFORCE(!framework::TensorContainsNAN(tensor),
+                   "Operator %s output Tensor %s contains NAN", op_type, name);
+  }
```

Comment:
What's the difference between `PADDLE_ENFORCE_NOTNANINF ` and `PADDLE_ENFORCE `?

Reply:
They can throw different exceptions.